### ISSUE TITLE
Integrate third party task services

### ISSUE THREAD

**Issue by ryanabx** [cosmic-utils/tasks#6]
I know cosmic-tasks is advertised as a *simple* task manager, but if you'd like ideas for future work, here's what I'd like to see:

Integration with Google/Microsoft/Apple task apis. This would probably be a larger undertaking, and I don't know the first thing about syncing tasks with IMAP or with google/microsoft/apple task apis

Here's a google link I found though about google tasks api
https://developers.google.com/tasks/reference/rest

**Comment by edfloreshz** [comment]
Thanks for opening this, right now I'm focusing on reaching a feature complete state but once that is done we could surely consider adding third party services.

I'm curious as to how COSMIC will handle third party accounts and if we could make use of that to handle task services here.

**Comment by ryanabx** [comment]
Yeah, if there's an account manager daemon or something similar, it'd be nice to have full integration with that.

**Comment by CristianKerr** [comment]
I would like to also confirm, that CalDav implementation would be awesome. Currently there is no proper task tool for linux (tasks, projects, tags, filtering) that have CalDav integration. There is e.g. Planify (most feature full) but they only integrate CalDav from Nextcloud. Or e.g. Errands, which works with any CalDav (even through eteSync) but is very basic (although they added tags recently, there is no filtering, just displaying them and adding/removing). Basically if you want to have full fledged experience on more devices you need to go for something like todoist which is proprietary though and you know nothing about what they do with your data.

**Comment by crcrewso** [comment]
> wesome. Currently there is no proper task tool for linux (tasks, projects, tags, filtering) that have CalDav integration. There is e.g. Planify (most feature full) but they only integrate CalDav from Nextcloud. Or e.g. Errands, which works with any CalDav (even through eteSync) but is very basic (although they added tags recently, there is no filtering, just displaying them and adding/removing

Right now I use thunderbird, not ideal but it works with all of my groupware accounts. Agreed that something more simple like this would be ideal.

**Comment by Duplicake-fyi** [comment]
would love a todoist integration

**Comment by BradNut** [comment]
Any idea if this will get integrated?

**Comment by edfloreshz** [comment]
Working on it: https://github.com/cosmic-utils/tasks/compare/main...feat/accounts
